otherlibs/win32unix/fsync.c

```diff
@@ -0,0 +1,83 @@
+/* Emulate fsync on platforms that lack it, primarily Windows and
+   cross-compilers like MinGW.
+
+   This is derived from sqlite3 sources.
+   http://www.sqlite.org/cvstrac/rlog?f=sqlite/src/os_win.c
+   http://www.sqlite.org/copyright.html
+
+   Written by Richard W.M. Jones <rjones.at.redhat.com>
+
+   Copyright (C) 2008-2012 Free Software Foundation, Inc.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
```

Comment:
This may not be compatible with the compiler's license (has no static linking exception).

Reply:
I can make that stronger - this code is *not* compatible with the compiler's contribution requirements. See https://github.com/ocaml/ocaml/pull/1148/commits/3d259e3239bc637a2eee0e12a2412553a97478cb. @rwmjones may be willing to contribute it to the compiler sources.